src/interface/data_utils.jl

```
@@ -19,18 +19,17 @@ MMI.int(::FI, x; args...) = CategoricalDistributions.int(x; args...)
 MMI.classes(::FI, x) = CategoricalDistributions.classes(x)
 
 # ------------------------------------------------------------------------
-# schema
-
+# `schema`
 MMI.schema(::FI, ::Val{:table}, X; kw...) = schema(X; kw...)
+MMI.schema(::FI, ::Val{:other}, X; kw...) = nothing
```

Line 24 imparts on the `schema` method used in implementations of the MLJ model interface (running in a session in which MLJBase is loaded) *different* behaviour than the `schema` method exported by MLJBase:

```julia
using MLJBase
julia> schema("abc")
ERROR: ArgumentError: Cannot inspect the internal scitypes of a non-tabular object. 
Stacktrace:
 [1] schema(X::String, #unused#::Val{:other})
   @ ScientificTypes ~/MLJ/ScientificTypes/src/schema.jl:115
 [2] schema(X::String)
   @ ScientificTypes ~/MLJ/ScientificTypes/src/schema.jl:112
 [3] top-level scope
   @ REPL[43]:1
 [4] top-level scope
   @ ~/.julia/packages/CUDA/YpW0k/src/initialization.jl:52

julia> MLJBase.MLJModelInterface.schema("abc")

julia> 
```
I suggest we remove line 24 to ensure same behaviour; `schema(X)` should throw an error if `X` is not a table.

---

Good Catch.